### PR TITLE
[WORKFLOWS-537] Creates `nf-synapse`, ports over DSL2 updated `nf-synstage` functionality

### PR DESCRIPTION
**Pull request by BWMac**
**Problem:**

Newer versions of Nextflow no longer support DSL1. As a result, we need to use workarounds to run workflows like `nf-synstage` on Nextflow Tower.

**Solution:**

Create a new centralized repository for Nextflow Workflows that interact with Synapse and create its first DSL2-compatible workflow which contains the functionality of `nf-synapse`.

Notes:
- This PR sets up the repository structure and provides documentation for how this repository should be used and contributed to in `README.md` and `CONTRIBUTING.md`.
- `nf-synapse` follows a similar pattern to `nf-synapse-challenge` in that it is a collection of workflows that are made up of individual modules.
- Updating the `nf-synstage` functionality for DSL2 required several changes to both the individual processes and the overall workflow.
- This new workflow has been tested using the `NF_SYNSTAGE` entrypoint locally and on Tower in all three possible file staging scenarios ([both Synapse and Seven Bridges files](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/5JOvBrZeCpFdhW), [Synapse only](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/4pJdbbdPw0lvya), and [Seven Bridges Only](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/j44KNKdbCM9WF)).

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -0,0 +1,39 @@
+profiles {
+	conda { process.conda = "$baseDir/environment.yml" }
```
Comment (BryanFauble) [inline nextflow.config:2]: nit: Consistent tabbing across the file

File: workflows/synstage.nf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+include { SYNAPSE_GET } from '../modules/synapse_get.nf'
+include { SEVENBRIDGES_GET } from '../modules/sevenbridges_get.nf'
+include { UPDATE_INPUT } from '../modules/update_input.nf'
+
+workflow SYNSTAGE {
+
+    // Warning if input file lacks URIs
+    if (params.synapse_uris.size() == 0 && params.sbg_uris.size() == 0) {
+    message = "The input file (${params.input}) does not contain any Synapse " +
```
Comment (BryanFauble) [inline workflows/synstage.nf]: For formatting does this allow you to indent by 1 from the `if` statement above?

**Review by BryanFauble — APPROVED** [review]
Excellent work!

File: nextflow.config (hunk 1 above)
Comment (BWMac) [reply nextflow.config:2]: standardized in latest commit

File: workflows/synstage.nf (hunk 2 above)
Comment (BWMac) [reply workflows/synstage.nf]: added indent in latest commit

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!

Lets make sure that we post in tower-news that the deprecation of nf-synstage is happening and that people should start using nf-synapse.

**Pull request merged**